### PR TITLE
Add export const config support and make withAmp a no-op

### PR DESCRIPTION
**Pull request by ijjk**
This adds support for `export const config = { amp: true | 'hybrid' }` and removes the need for `withAmp(() => <p>hello world</p>)`

### CODE REVIEW

File: packages/next-server/lib/amp.ts [hunk 1]
```diff
@@ -15,16 +15,9 @@ export function useAmp() {
   return isAmp(ampMode) // && ampMode.hasQuery
 }
 
+/**
+ * @deprecated This is no longer required, use export const config = { amp: true }
```
Comment (timneutkens) [inline packages/next-server/lib/amp.ts]: This is a breaking change right 🤔

Comment (ijjk) [reply packages/next-server/lib/amp.ts]: Yeah, should we just remove it?

Comment (timneutkens) [reply packages/next-server/lib/amp.ts]: Can we make it not breaking?

**Comment by Janpot** [comment]
Does it need to be explicitly exported, or would it work when it's a static property as well?
like:
```js
export default function MyPage () {
  // ...
}
MyPage.amp = true;
```

**Comment by timneutkens** [comment]
It needs to be an explicit export so that we can optimize it.

File: packages/next-server/lib/amp.ts (hunk 1 above)
Comment (ijjk) [reply packages/next-server/lib/amp.ts]: Added backwards compat with `withAmp`

**Pull request merged**